src/Microsoft.AspNetCore.Razor.Language/RequiredAttributeDescriptorComparer.cs

```diff
@@ -63,19 +63,14 @@ private RequiredAttributeDescriptorComparer(bool caseSensitive = false)
                 descriptorX.ValueComparison == descriptorY.ValueComparison &&
                 string.Equals(descriptorX.Name, descriptorY.Name, _stringComparison) &&
                 string.Equals(descriptorX.Value, descriptorY.Value, StringComparison.Ordinal) &&
-                string.Equals(descriptorX.DisplayName, descriptorY.DisplayName, StringComparison.Ordinal) &&
-                Enumerable.SequenceEqual(descriptorX.Diagnostics, descriptorY.Diagnostics);
+                string.Equals(descriptorX.DisplayName, descriptorY.DisplayName, StringComparison.Ordinal);
```

Comment:
ditto with `_stringComparison`, I wont leave any more comments like this. Looks like it's also a thing in TagHelperDescriptorComparer.